src/Branching/perform-migration-from-svn-to-git.md

```diff
@@ -0,0 +1,184 @@
+
+```
+git svn clone ["SVN repo URL"] --prefix=svn/ --no-metadata --trunk=/trunk --branches=/branches --tags=/tags  --authors-file "authors-transform.txt" c:\mytempdir
+```
+
+> [!NOTE]
+>
+> This command can take a few minutes to several hours depending on the size of the SVN repository. Upon completion, you will have a Git checkout of your repository.
+
+### Convert version control-specific configurations
+
+If your SVN repo was using svn:ignore properties, you can  convert this to a .gitignore file using:
+```
+cd c:\mytempdir
+git svn show-ignore > .gitignore
+git add .gitignore
+git commit -m 'Convert svn:ignore properties to .gitignore.'
+```
+> [!TIP]
+>
+> Read more about .gitignore [Ignore file changes with Git](
+https://docs.microsoft.com/en-us/vsts/git/tutorial/ignore-files?tabs=visual-studio)
+
+### Push repository to a bare git repository
+
+In this step, you will create a bare repository and make its default branch match SVN's trunk branch name.
+
+1. Create a bare Git repository
+
+```
+git init --bare c:\new-bare.git
+cd c:\new-bare.git
+git symbolic-ref HEAD refs/heads/trunk
+```
+2. Push the local Git repository to the new bare Git repository
+```
+cd c:\mytempdir
+git remote add bare c:\new-bare.git
+git config remote.bare.push 'refs/remotes/*:refs/heads/*'
+git push bare
+```
+3. Rename "trunk" branch to "master"
+Your main development branch will be named “trunk” which matches the name it was in Subversion. You’ll want to rename it to Git’s standard “master” branch using:
+```
+cd c:\new-bare.git
+git branch -m trunk master
+```
+4. Clean up branches and tags
+git-svn makes all of Subversions tags into very-short branches in Git of the form “tags/name”. You’ll want to convert all those branches into actual Git tags or delete them.
+
+### Migrate SVN tags to be Git tags
+
+```
+cd c:\new-bare.git
+git for-each-ref refs/remotes/tags | cut -d / -f 4- | grep -v @ | while read tagname; do git tag "$tagname" "tags/$tagname"; git branch -r -d "tags/$tagname"; done
+
+```
+
+### Advanced migrations
+
+### Create all the SVN branches as proper Git branches
+
+While it's easy to create all SVN branches as a proper Git branches, we do recommend that you evaluate the following points before you proceed:
+
+- In case of Feature branches – can you wait until they integrate to the trunk before migrating?
+
+- In case of Release branches – does it make sense to keep SVN around for servicing?  If you migrate feature branches, are you prepared to service branches out of Git?
+
+If you still want to migrate existing branches, then running the following command will help
+```
+git for-each-ref refs/remotes | cut -d / -f 3- | grep -v @ | while read branchname; do git branch "$branchname" "refs/remotes/$branchname"; git branch -r -d "$branchname"; done
```

Comment:
The Powershell equivalent for creating svn branches as proper git branches is as follows:

git for-each-ref refs/remotes | % { $_.Substring($_.IndexOf('refs/remotes/')+13) } | % { git branch "$_" "refs/remotes/$_"; git branch -r -d "$_"; }